lib/cocina/models/validators/language_tag_validator.rb

```diff
@@ -17,30 +17,56 @@ def initialize(clazz, attributes)
         def validate
           return unless meets_preconditions?
 
-          return if valid_language_tag?
+          return if invalid_files.empty?
 
-          raise ValidationError, 'The provided language tag is not valid according to RFC 4646: ' \
-                                 "#{attributes[:languageTag]}"
+          raise ValidationError, 'Some files have invalid language tags according to RFC 4646: ' \
+                                 "#{invalid_filenames_with_language_tags.join(', ')}"
         end
 
         private
 
         attr_reader :clazz, :attributes
 
         def meets_preconditions?
-          file? && attributes[:languageTag].present?
+          dro?
         end
 
-        def file?
-          (clazz::TYPES & File::TYPES).any?
+        def dro?
+          (clazz::TYPES & DRO::TYPES).any?
         rescue NameError
           false
         end
 
-        def valid_language_tag?
-          parsed_tag = I18n::Locale::Tag::Rfc4646.tag(attributes[:languageTag])
+        def valid_language_tag?(file)
+          # I18n::Locale::Tag::Rfc4646.tag will return an instance of I18n::Locale::Tag::Rfc4646 (with fields like language, script,
+          # region) for strings that can be parsed according to RFC 4646, and nil for strings that do not conform to the spec.
+          I18n::Locale::Tag::Rfc4646.tag(file[:languageTag]).is_a?(I18n::Locale::Tag::Rfc4646)
```

Comment:
Why not just `I18n::Locale::Tag::Rfc4646.tag(file[:languageTag])` as a nil check?

Reply:
oh, good question, no particular reason.  i think use of `.is_a?` is partially an artifact of playing with this on the console to see what the return type was for valid and invalid strings.

not having thought about it much, i may very slightly prefer this, because it's more explicit about the expected return value from the parser, but i don't feel strongly at all and am totally happy to just do a nil check if you'd prefer.

Reply:
It seems like we care more about the return value of `I18n::Locale::Tag::Rfc4646#tag` being non-`nil` than being a legit `I18n::Locale::Tag::Rfc4646` struct instance (which we throw away). Do you think `#present?` would reflect better that this is what we care about?

Reply:
i think i still slightly prefer the `is_a?`, but since it's not a strong preference at all, i'm fine being outvoted, changed to `#present?` and re-pushed 🙂 (i do agree that we care more about non-nil here -- i think my pref mostly stems from the implicit guidance the `is_a?` provided if a cocina-models consumer wanted to parse the field, but the comment makes that same guidance more explicit, so the `is_a?` then feels even less necessary)